modules/MachineLearning/MaliciousMacroBot.py

```diff
@@ -0,0 +1,60 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+from __future__ import division, absolute_import, with_statement, print_function, unicode_literals
+
+__authors__ = "Austin West"
+__license__ = "MPL 2.0"
+
+TYPE = "MachineLearning"
+NAME = "MaliciousMacroBot"
+REQUIRES = ['libmagic']
+DEFAULTCONF = {
+    'ENABLED': True
+    }
+
+try:
+    from mmbot import MaliciousMacroBot
+    has_mmbot = True
+except:
+    print("mmbot module not installed...")
+    has_mmbot = False
+
+
+def check(conf=DEFAULTCONF):
+    if not conf['ENABLED']:
+        return False
+    if not has_mmbot:
+        return False
+    return True
+
+
+def scan(filelist, conf=DEFAULTCONF):
+    results = []
+
+    mmb = MaliciousMacroBot()
+    mmb.mmb_init_model()
+
+    for fname in filelist:
+        # Ensure libmagic returns results
+        if REQUIRES[0] is not None:
+            # only run the analytic if it is an Office document
+            if 'Microsoft' in _get_libmagicresults(REQUIRES[0][0], fname):
```

Comment:
Do you know if the tool supports other document file formats like OpenDocument? It seems like they might get excluded based on this check.

Reply:
https://wiki.openoffice.org/wiki/Documentation/FAQ/Macros/Can_I_use_my_Microsoft_Office_macros%3F

Reply:
You are right. From what I can tell, this tool will run on any file but will return a "benign" prediction for anything that is not an office document. However, I didn't test it with any Open Office documents.